pkg/api/kubegraph/analysis/hpa.go

```diff
@@ -0,0 +1,178 @@
+package analysis
+
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/kubernetes/pkg/apis/extensions"
+
+	graphapi "github.com/gonum/graph"
+	"github.com/gonum/graph/path"
+
+	osgraph "github.com/openshift/origin/pkg/api/graph"
+	"github.com/openshift/origin/pkg/api/kubegraph"
+	kubenodes "github.com/openshift/origin/pkg/api/kubegraph/nodes"
+	deploygraph "github.com/openshift/origin/pkg/deploy/graph"
+	deploynodes "github.com/openshift/origin/pkg/deploy/graph/nodes"
+)
+
+const (
+	// HPAMissingScaleRefError denotes an error where a Horizontal Pod Autoscaler does not have a reference to an object to scale
+	HPAMissingScaleRefError = "HPAMissingScaleRef"
+	// HPAMissingCPUTargetError denotes an error where a Horizontal Pod Autoscaler does not have a CPU target to scale by.
+	// Currently, the only supported scale metric is CPU utilization, so without this metric an HPA is useless.
+	HPAMissingCPUTargetError = "HPAMissingCPUTarget"
+	// HPAOverlappingScaleRefWarning denotes a warning where a Horizontal Pod Autoscaler scales an object that is scaled by some other object as well
+	HPAOverlappingScaleRefWarning = "HPAOverlappingScaleRef"
+)
+
+// FindHPASpecsMissingCPUTargets scans the graph in search of HorizontalPodAutoscalers that are missing a CPU utilization target.
+// As of right now, the only metric that HPAs can use to scale pods is the CPU utilization, so if a HPA is missing this target it
+// is effectively useless.
+func FindHPASpecsMissingCPUTargets(graph osgraph.Graph, namer osgraph.Namer, patchCommandName string) []osgraph.Marker {
```

Comment:
> I think that in a place where we're giving a user who presumably doesn't really know what went wrong or how to fix it a command that they will inevitably copy-paste we should try as hard as we can to ensure that the command that they copy-paste will work 100%

It looks ugly in the other spots too.  I don't think its likely enough to cause a problem that I want to plumb it through like this.  I don't like it anywhere and you're already making assumptions about the shape of the command to recommend patch.

This is a spot where a global makes sense ("which process am I?"), but we aren't going to add it now.